src/v/storage/types.h

```diff
@@ -37,7 +37,7 @@ class snapshotable_stm {
     // offset already exists
     virtual ss::future<> ensure_snapshot_exists(model::offset) = 0;
     // hints stm_manager that now it's a good time to make a snapshot
-    virtual ss::future<> make_snapshot() = 0;
```

Comment:
can we leave this function signature as-is and launch background fibers in `stm_manager`?

Reply:
changing the function signature reflects the idea that this is a hint and force implementer to perform operation in background. my idea was to make it less error prone

Reply:
makes sense